src/portfft/utils.hpp

```diff
@@ -129,17 +134,20 @@ IdxGlobal factorize_input_impl(IdxGlobal factor_size, F&& check_and_select_targe
  * implementations. The function should accept factor size and whether it would be have a BATCH_INTERLEAVED layout or
  * not as an input, and should return a boolean indicating whether or not the factor size can fit in any of the
  * implementation.
+ * @param Whether or not the factorization was successful
  */
 template <typename F>
-void factorize_input(IdxGlobal input_size, F&& check_and_select_target_level) {
+bool factorize_input(IdxGlobal input_size, F&& check_and_select_target_level) {
```

Comment:
I don't really understand this approach here.

We want to be able to handle inputs sizes where the factors include large primes (that might then be tackled with Bluestein), so shouldn't this be able to handle large primes too? The pure-Bluestien case being that there is a single large prime factor.

Reply:
If the factorization fails because it encounters a prime number, this function returns false, hence the returning of  boolean,
Does this answer your question ?

Reply:
No - I got that.
I don't understand why we want to branch when we encounter a large prime. I'll try and illustrate with a large 1D dft being decomposed into factors that can be computed with smaller DFTs:

1. dftSize = 256 * 256 * 256
Therefore we dispatch as three workgroup implementations: WG(256), WG(256), WG(256).

2. dftSize = 947 (a big prime!).
Therefore we dispatch as Bluestein(947). This can further be padded, decomposed and dispatched into smaller DFTs.

3. dftSize = 256 * 256 * 113 (prime).
Therefore we dispatch as WG(256), WG(256), Bluestein(113). The Bluestein part can be further padded, decomposed, dispatched etc.

Case 1 is what normally happens (-ish). 

Case 2 and 3 should be possible to handle with the same code, which means having `check_and_select_target_level` also handle Bluestein. Consequently, you don't need to return anything special when a large prime is encountered. 

At the moment, as I understand it, in case 2 the code decides its a big prime, and therefore gives it different treatement to case 3, despite both using Bluestein?

Reply:
I get your question now.
No we do not dispatch to bluestein at a per factor level. Rather if any size has a prime size factor, entire size gets padded to a favourable size.

What you suggest cannot be possible at the moment, (we cannot have a global level factor (or what uses the global implementation) from a global implementation). 
Yes I am aware this is suboptimal, but this is what I suppose we will be going with for now. Later, I suppose (or rather that's what I suppose should happen), we will implement subgroup and workgroup level bluestein algorithm , so that cases like the one in your example , we would not need to pad the entire size, but only a certain factor. We would only pad  if a prime size cannot fit in workgroup implementation, once we have subgroup / wg level bluestein.

Reply:
Pitty its not possible right now - I think some of the comments in this PR are a result of you working around the existing implementation.

Thanks for explaining.